src/actions/sources/select.js

```diff
@@ -137,6 +143,12 @@ export function selectLocation(location: Location) {
 
     dispatch(setSymbols(loadedSource.id));
     dispatch(setOutOfScopeLocations());
+
+    // If a new source is selected update the file search results
+    const newSource = getSelectedSource(getState());
+    if (!currentSource || (newSource && currentSource.id != newSource.id)) {
```

Comment:
maybe we can be even simpler. This will say, there had to have been a current source and it can't be the same as the new one. I don't know if the IDs matter

```js
if (currentSource && currentSource !== newSource) {
}
```

Reply:
I tried it this way but it would sometimes return true even when the source did not change, such as when clicking on the editor tab or source file of the file that is already active.  The objects were different even though they were pointing to the same source.
![screen shot 2018-07-05 at 10 44 21 am](https://user-images.githubusercontent.com/5035181/42339113-e0930e32-8040-11e8-9615-8410f5503908.png)

Reply:
nice find. try changing this and see if that is still the case. If it is, lets figure out why the object is changing!

```diff

diff --git a/src/reducers/sources.js b/src/reducers/sources.js
index bb18ebc..a74bf7c 100644
--- a/src/reducers/sources.js
+++ b/src/reducers/sources.js
@@ -416,15 +416,7 @@ export const getSelectedSource = createSelector(
       return;
     }

-    const source = sources[selectedLocation.sourceId];
-
-    // TODO: remove this when the immutable refactor lands in m-c
-    if (isTesting) {
-      const testSource: any = { ...source, get: field => source[field] };
-      return testSource;
-    }
-
-    return source;
+    return sources[selectedLocation.sourceId];
   }
 );
```

Reply:
just noticed this should be ` if (isTesting()) {`

Reply:
I added the parentheses and that fixed it and now comparing for the same object works :)